server/server/Controllers/AdminController.cs

```diff
@@ -0,0 +1,102 @@
+﻿using MailKit.Security;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using MimeKit;
+using server.DBRepositories;
+using server.Models;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace server.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize(Roles = "Admin")]
+    public class AdminController: Controller
+    {
+
+        private UnitOfWork db;
+        public AdminController(ApplicationContext context)
+        {
+            db = new UnitOfWork(context);
+        }
+
+        static void SendEmail(string from, string to, string name, string cakeDescription, string login, string password)
+        {       
+            MimeMessage message = new MimeMessage();
+            message.From.Add(new MailboxAddress("MasterCake", from));
+            message.To.Add(new MailboxAddress("", to));
+            message.Subject = "New cake in MasterCake";
+            message.Body = new BodyBuilder() { HtmlBody = 
+                "<div " +
+                    "<div style=\"padding-bottom: 5%;\"><em>Dear " + name  + ",</em></div>" +
+                    "<b style=\"color: purple;\">We have a new cake in our catalog especially for you!</b>" +
+                    "<div style=\"padding: 10% 0;\"><em>" + cakeDescription + "</em></div>" + 
+                    "<b>Hurry to order!</b>" +
+                    "<div style=\"padding-top: 20%;\"><em>With love, your MasterCake)</em></div>" +
+                "</div>" }.ToMessageBody(); 
+
+            using (MailKit.Net.Smtp.SmtpClient client = new MailKit.Net.Smtp.SmtpClient())
+            {
+                client.Connect("smtp.mail.ru", 25, SecureSocketOptions.StartTls);
+                client.Authenticate(login, password);
+                client.Send(message);
+
+                client.Disconnect(true);
+            }
+        }
+
+        [HttpPost]
+        [Authorize]
+        public IActionResult Post(AddedCake cake)
+        {
+            char separator = System.Globalization.CultureInfo.CurrentCulture.NumberFormat.CurrencyDecimalSeparator[0];
+            string stringPrice = cake.Price.Replace('.', separator);
+            decimal price = Convert.ToDecimal(stringPrice);
```

Comment:
In case if price contains invalid string exception will be throw. Better to use decimal.TryParse